frontend/libs/studio-components/src/components/StudioCodelistEditor/StudioCodeListEditor.test.tsx

```diff
@@ -506,6 +502,105 @@ describe('StudioCodeListEditor', () => {
   });
 
   describe('Type handling', () => {
+    it('Renders type selector when code list is empty', () => {
+      renderCodeListEditor({ codeList: [] });
+      const typeSelector = screen.getByRole('combobox', {
+        name: texts.typeSelector.label,
+      });
+      expect(typeSelector).toBeInTheDocument();
+    });
+
+    it('Does not render type selector when code list is already populated', () => {
+      renderCodeListEditor();
+      const typeSelector = screen.queryByRole('combobox');
+      expect(typeSelector).not.toBeInTheDocument();
+    });
+
+    it("Creates an empty string item when string is selected and 'Add new' is pressed", async () => {
+      const user = userEvent.setup();
+      renderCodeListEditor({ codeList: [] });
+
+      const typeSelector = screen.getByRole('combobox');
+      const stringOption = screen.getByRole('option', {
+        name: texts.typeSelector.stringOption,
+      });
+      const addButton = screen.getByRole('button', { name: texts.add });
+
+      await user.selectOptions(typeSelector, stringOption);
+      await user.click(addButton);
+
+      expect(onAddOrDeleteItem).toHaveBeenCalledWith([emptyStringItem]);
```

Comment:
Can we also have a check to expect that it has only been called one time? 😄 
```tsx
      expect(onAddOrDeleteItem).toHaveBeenCalledTimes(1);
```

This also goes for the tests below this one 😄

Reply:
Fixed.